### PR TITLE
Issue #315: Add the "Entity view mode" module to core

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes https://github.com/backdrop/backdrop-issues/issues/315

### CODE REVIEW

File: core/modules/system/system.api.php [hunk 1]
```diff
@@ -59,24 +59,6 @@ function hook_hook_info_alter(&$hooks) {
...
- *
- * @param string $view_mode
- *   The view_mode that is to be used to display the entity.
- * @param array $context
- *   Array with contextual information, including:
- *   - entity_type: The type of the entity that is being viewed.
- *   - entity: The entity object.
- *   - langcode: The langcode the entity is being viewed in.
- */
-function hook_entity_view_mode_alter(&$view_mode, $context) {
```
Comment (jenlampton) [inline core/modules/system/system.api.php:72]: Removing a hook is an API break. We'll need to find a way for this to remain.

**Comment by jenlampton** [comment]
I think this is a fantastic start @docwilmot and I've added this as needs-review to today's dev meeting agenda.  Approach looks solid to me 👍

File: core/modules/system/system.api.php (hunk 1 above)
Comment (docwilmot) [reply core/modules/system/system.api.php:72]: Its not removed. This just moved the API *documentation* from `system.api.php` to `entity.api.php`.

Comment (jenlampton) [reply core/modules/system/system.api.php:72]: Ah!  nevermind then :)

File: core/modules/entity/entity.api.php [hunk 2]
```diff
@@ -423,3 +423,157 @@ function hook_entity_prepare_view($entities, $type) {
...
+ *
+ * @param string $view_mode
+ *   The view_mode that is to be used to display the entity.
+ * @param array $context
+ *   Array with contextual information, including:
+ *   - entity_type: The type of the entity that is being viewed.
+ *   - entity: The entity object.
+ *   - langcode: The langcode the entity is being viewed in.
+ */
+function hook_entity_view_mode_alter(&$view_mode, $context) {
```
Comment (jenlampton) [inline core/modules/entity/entity.api.php:499]: yep, bingo. Sorry I missed that!

**Comment by quicksketch** [comment]
https://github.com/backdrop/backdrop-issues/issues/315 has been fixed in separate PRs. Woot!

**Pull request closed without merging**